frontend/src/metabase/actions/containers/ActionCreator/ActionCreator.tsx

```diff
@@ -98,11 +102,25 @@ function ActionCreator({
     renderEditorBody,
   } = useActionContext();
 
-  useBeforeUnload(isDirty);
-
   const [isSaveModalShown, setShowSaveModal] = useState(false);
 
   const isEditable = isNew || (model != null && model.canWriteActions());
+  const [actionToSubmit, scheduleSubmitAction] = useState<WritebackAction>();
+
+  const showUnsavedChangesWarning = isEditable && isDirty && !actionToSubmit;
+
+  useBeforeUnload(!route && showUnsavedChangesWarning);
```

Comment:
I'm not sure why can't we just use `showUnsavedChangesWarning` here to simplify the logic. Since we're gonna set `useBeforeUnload(true)` inside `LeaveConfirmationModal` anyway.

If that happens we'll just call `useBeforeUnload(true)` 2 times, which won't make any difference in the behavior. Having `!route` here makes me feel that we don't want to call `useBeforeUnload` in some situations, but in reality, it will be called anyway but in another place.

What do you think about simplifying this condition?

Reply:
> I'm not sure why can't we just use `showUnsavedChangesWarning` here to simplify the logic. Since we're gonna set `useBeforeUnload(true)` inside `LeaveConfirmationModal` anyway.

I wanted to explicitly make it the reverse of when `LeaveConfirmationModal` is enabled.
This way, if some day we can get rid of [that condition](https://github.com/metabase/metabase/pull/34192/files#r1339691613), it will be easier for the person making the change to determine whether `useBeforeUnload` usage can be removed.

> If that happens we'll just call useBeforeUnload(true) 2 times, which won't make any difference in the behavior.

It can make difference in tests: if someone uses `toHaveBeenCalledTimes(1)` they're in for a debugging journey.

> What do you think about simplifying this condition?

I'd keep it. Only 1 `beforeunload` handler is expected on the page.

Reply:
sounds good to me.